ddclient

```diff
@@ -2298,57 +2268,81 @@ sub get_ip {
 }
 
 ######################################################################
-## is_ipv4() validates if string is valid IPv4 address and only
-## a valid IPv4 address (no preceding or trailing spaces/characters)
+## is_ipv4() validates if string is valid IPv4 address and only a
+## valid address with no preceding or trailing spaces/characters
+## and no embedded leading zeros.
 ######################################################################
 sub is_ipv4 {
     my ($value) = @_;
-    return (length($value // '') != 0) && ($value eq (ipv4_match($value) // ''));
+    return (length($value // '') != 0) && ($value eq (extract_ipv4($value) // ''));
 }
 
 ######################################################################
-## ipv4_match() extracts the first valid IPv4 address from given string
+## extract_ipv4() extracts the first valid IPv4 address from given string.
+## Accepts leading zeros in the address but removes them in returned value
 ######################################################################
-sub ipv4_match {
-    (shift // '') =~ /\b((?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?))\b/ai;
-    return $1;
+sub extract_ipv4{
+    if ((shift // '') !~ /\b((?:(?<octet>25[0-5]|2[0-4]\d|[01]?\d\d?)\.){3}
+                         (?&octet))\b/xa) {
```

Comment:
The regex is short enough that you can remove the line break and it'll still fit in <100 columns:

```suggestion
    if ((shift // '') !~ /\b((?:(?<octet>25[0-5]|2[0-4]\d|[01]?\d\d?)\.){3}(?&octet))\b/a) {
```

If you want to split it across multiple lines so that you can embed comments (I don't think they're necessary for this regex), it would probably be more readable to store the regex in a variable:

```suggestion
    my $re = qr/\b(
            (?:(?<octet>25[0-5]|2[0-4]\d|[01]?\d\d?)\.){3}  # first three octets
            (?&octet)                                       # last octet
        )\b/xa;
    if ((shift // '') !~ $re) {
```